source/server/configuration_impl.h

```diff
@@ -162,6 +191,17 @@ template <class T> class RegisterNetworkFilterConfigFactory {
 };
 
 /**
+ * @see HttpTracerFactory.
+ */
+template <class T> class RegisterHttpTracerFactory {
+public:
+  RegisterHttpTracerFactory() {
+    static T instance;
```

Comment:
In addition to the above `static std::list`, this should also be created on use. For reference:

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables
https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use